### ISSUE TITLE
Somfy Window Coverings

### ISSUE THREAD

**Issue by A5HRAJ** [SphtKr/homebridge-zway#43]
Somfy's ZRTSI is a bridge controller that receives Z-Wave commands and outputs RTS commands for up to 16 Somfy window coverings. It can only output three commands: up, down, and stop\* — but Z-Way detects multilevel control and switch status, both of which should not exist. If a Z-Wave value level command besides 0 or 99 is sent to ZRTSI, the command is ignored. If the status of a window covering is polled, ZRTSI reports the last command that was sent to _any_ shade. E.g., if all 16 shades are closed, and I send a command to just open shade 1, ZRTSI will report that all of the shades have been opened even though 2-16 are still closed.

Is there some way to configure your module to only expose up, down, and stop commands? Currently, HomeKit lets me set the shade level (which does nothing) and doesn't have a way to stop the shade.

Thanks!
Ash

*If the stop command is sent while a shade is already stopped, the shade will move to a preconfigured preset position. This logic exists in the shade itself and the event is not reported to ZRTSI.

**Comment by SphtKr** [comment]
Hmm...I thought I had support for "binary switch" window coverings before, but I don't see it now. Either I'm misremembering or I took it out when I added multilevel support. Anyway...

Can you send me a your `/devices` JSON output? I can use that to figure out which kind of device needs the command sent to.

**Comment by A5HRAJ** [comment]
```
{
    "data":{
        "structureChanged":true,
        "updateTime":1469356292,
        "devices":[
            {
                "creationTime":1469355042,
                "creatorId":7,
                "deviceType":"battery",
                "h":-592588978,
                "hasHistory":false,
                "id":"BatteryPolling_7",
                "location":0,
                "metrics":{
                    "probeTitle":"Battery",
                    "scaleTitle":"%",
                    "title":"Battery digest 7"
                },
                "permanently_hidden":false,
                "probeType":"",
                "tags":[

                ],
                "visibility":true,
                "updateTime":1469355042
            },
            {
                "creationTime":1469355662,
                "creatorId":1,
                "deviceType":"switchMultilevel",
                "h":1107079067,
                "hasHistory":false,
                "id":"ZWayVDev_zway_3-0-38",
                "location":0,
                "metrics":{
                    "icon":"blinds",
                    "title":"Somfy Shade 1",
                    "level":1
                },
                "permanently_hidden":false,
                "probeType":"motor",
                "tags":[

                ],
                "visibility":true,
                "updateTime":1469355773
            },
            {
                "creationTime":1469355688,
                "creatorId":1,
                "deviceType":"switchBinary",
                "h":1107079066,
                "hasHistory":false,
                "id":"ZWayVDev_zway_3-0-37",
                "location":0,
                "metrics":{
                    "icon":"switch",
                    "title":"Somfy Switch 1",
                    "level":"off"
                },
                "permanently_hidden":false,
                "probeType":"",
                "tags":[

                ],
                "visibility":true,
                "updateTime":1469355773
            }
        ]
    },
    "code":200,
    "message":"200 OK",
    "error":null
}
```

For some reason, each virtual node shows up as two elements. "Somfy Shade 1" and "Somfy Switch 1" operate the same shade. Both of them can raise and lower the shade, but only "Somfy Shade 1" can stop the shade (using the command StopLevelChange).

**Comment by SphtKr** [comment]
Believed fixed in 969d905, release 0.5.0-alpha4. Reopen if still an issue!
